src/util/helpers.ts

```diff
@@ -272,26 +274,38 @@ function _fetchBtcTxs(addresses, cb, txs=[], offset=0, isFirstCall=true) {
             if (!ftx.confirmed) {
                 ftx.timestamp = -1;
             }
-            formattedTxs.push(ftx);
+
+            // Only add the transaction if its hash is not already in the array.
+            // NOTE: There may be an edge case. I noticed in one case we got
+            // a result saying `vout_sz=2` but which only had one output in its array...
+            let shouldInclude = true;
+            txs.forEach((_tx) => {
+                if (_tx.id === ftx.id) {
+                    shouldInclude = false;
+                }
+            })
+            if (shouldInclude) {
```

Comment:
Same as above. This could be: 

```ts
txs.every(_tx => _tx.id !== ftx.id)
```

Reply:
[2211973](https://github.com/GridPlus/lattice-manager/pull/219/commits/221197375f71b9b0441d0868439625d6cc3ddddb)